### PR TITLE
feat: include server streaming in rest descriptor

### PR DESCRIPTION
**Pull request by noahdietz**
Enables generation of a REST descriptor for server-stream RPCs in order to support REGAPIC streaming.

### CODE REVIEW

File: src/Generation/ResourcesGenerator.php [hunk 1]
```diff
@@ -210,7 +210,7 @@ public static function generateRestConfig(ServiceDetails $serviceDetails, Servic
     private static function compileRestConfigInterfaces(ServiceDetails $serviceDetails, ServiceYamlConfig $serviceYamlConfig)
     {
         return $serviceDetails->methods
-            ->filter(fn ($method) => !is_null($method->httpRule) && !$method->isStreaming() && !$method->isMixin())
+            ->filter(fn ($method) => !is_null($method->httpRule) && ($method->isServerStreaming() || !$method->isStreaming()) && !$method->isMixin())
```
Comment (bshaffer) [inline src/Generation/ResourcesGenerator.php:213]: Why is this necessary, when `$this->methodType === static::SERVER_STREAMING` is included in the function `isStreaming`?

File: src/Generation/MethodDetails.php [hunk 2]
```diff
@@ -536,6 +536,11 @@ public function isStreaming(): bool
             $this->methodType === static::CLIENT_STREAMING;
     }
 
+    public function isServerStreaming(): bool
+    {
+        return $this->methodType === static::SERVER_STREAMING;
+    }
```
Comment (bshaffer) [inline src/Generation/MethodDetails.php:542]: I think this function may be unnecessary...

File: src/Generation/ResourcesGenerator.php (hunk 1 above)
Comment (noahdietz) [reply src/Generation/ResourcesGenerator.php:213]: We _want_ to enable rest config generation `SERVER_STREAMING` but not `CLIENT_STREAMING` or `BIDIRECTIONAL_STREAMING` which `isStreaming` returns `true` for.

File: src/Generation/MethodDetails.php (hunk 2 above)
Comment (noahdietz) [reply src/Generation/MethodDetails.php:542]: I explained in the previous comment relating to this. We want to generate a rest descriptor config for server streaming, but _not_ the other two types of streaming (because we haven't been).

File: src/Generation/ResourcesGenerator.php (hunk 1 above)
Comment (bshaffer) [reply src/Generation/ResourcesGenerator.php:213]: Oops sorry, I missed the good ol `!` at the start of `$this->isStreaming()`

**Review by bshaffer — APPROVED** [review]
